File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0016-some-every/type/TItemPrice.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tItemPrice"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TItemPrice extends com.gs.dmn.runtime.DMNType {
    static TItemPrice toTItemPrice(Object other) {
        if (other == null) {
            return null;
        } else if (TItemPrice.class.isAssignableFrom(other.getClass())) {
            return (TItemPrice)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TItemPriceImpl result_ = new TItemPriceImpl();
            result_.setItemName((String)((com.gs.dmn.runtime.Context)other).get("itemName"));
            result_.setPrice((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("price"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTItemPrice(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TItemPrice.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("itemName")
    String getItemName();

    @com.fasterxml.jackson.annotation.JsonGetter("price")
    java.math.BigDecimal getPrice();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("itemName", getItemName());
        context.put("price", getPrice());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TItemPrice other = (TItemPrice) o;
        if (this.getItemName() != null ? !this.getItemName().equals(other.getItemName()) : other.getItemName() != null) return false;
        if (this.getPrice() != null ? !this.getPrice().equals(other.getPrice()) : other.getPrice() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getItemName() != null ? this.getItemName().hashCode() : 0);
        result = 31 * result + (this.getPrice() != null ? this.getPrice().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("itemName=" + getItemName());
        result_.append(", price=" + getPrice());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0016-some-every/type/TItemPrice.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tItemPrice"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TItemPriceImpl::class)
interface TItemPrice : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("itemName")
    val itemName: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("price")
    val price: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("itemName", this.itemName)
        context.put("price", this.price)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TItemPrice
        if (if (this.itemName != null) this.itemName != other.itemName else other.itemName != null) return false
        if (if (this.price != null) this.price != other.price else other.price != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.itemName != null) this.itemName.hashCode() else 0)
        result = 31 * result + (if (this.price != null) this.price.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("itemName=" + itemName)
        result_.append(", price=" + price)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTItemPrice(other: Any?): TItemPrice? {
            if (other == null) {
                return null
            } else if (other is TItemPrice?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TItemPriceImpl()
                result_.itemName = other.get("itemName") as String?
                result_.price = other.get("price") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTItemPrice(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TItemPrice::class.java.getSimpleName()))
            }
        }
    }
}
